core/src/main/java/ai/timefold/solver/core/config/util/ConfigUtils.java

```diff
@@ -204,6 +204,20 @@ public static <T> List<T> inheritMergeableListProperty(List<T> originalList, Lis
         }
     }
 
+    public static <T> List<T> inheritUniqueMergeableListProperty(List<T> originalList, List<T> inheritedList) {
+        if (inheritedList == null) {
+            return originalList;
+        } else if (originalList == null) {
+            // Shallow clone due to modifications after calling inherit
+            return new ArrayList<>(inheritedList);
+        } else {
+            // The inheritedList should be before the originalList
+            List<T> mergedList = new ArrayList<>(inheritedList);
+            mergedList.addAll(originalList.stream().filter(v -> !mergedList.contains(v)).toList());
+            return mergedList;
```

Comment:
I think it'd read easier if it were instead:

    Set<T> mergedSet = new LinkedHashSet<>(originalList);
    mergedSet.addAll(inheritedList);
    return new ArrayList<>(mergedSet);

Maintains insertion order, doesn't need to use any streams, and absolute efficiency is not required here.